### ISSUE TITLE
Support releases without dune-project files

### ISSUE THREAD

**Issue by samoht** [ocamllabs/dune-release#437]
It's convenient to add a `dune-project` file at the root a non-dune project to ease release (good example: https://github.com/mirage/ocaml-solo5) - it'd be nice if the project name could also just be given on the command-line (and in that case, to not complain if a `dune-project` file is not present).

**Comment by NathanReb** [comment]
Indeed! We'll remove the need for a dune-project file in `2.0.0` as it is only used to determine the tarball name. We'll use the name of the repository in `dev-repo` for this as the repo ultimately is what the tarball contains.
